src/NuGet.Core/NuGet.PackageManagement/NuGet.PackageManagement.csproj

```diff
@@ -55,5 +55,8 @@
     <InternalsVisibleTo Include="DynamicProxyGenAssembly2" />
     <InternalsVisibleTo Include="NuGet.PackageManagement.Test" />
     <InternalsVisibleTo Include="NuGet.SolutionRestoreManager.Test" />
+    <InternalsVisibleTo Include="NuGet.CommandLine.FuncTest" />
+    <InternalsVisibleTo Include="Msbuild.Integration.Test" />
+    <InternalsVisibleTo Include="NuGet.Commands.FuncTest" />
```

Comment:
`InternalsVisibleTo` removes compiler protections for keeping a "clean" software architecture. From what I can tell, it's added in this PR just to validate strings, but now these test assemblies have access to all of this assembly's internals, so future modifications to the tests could unknowingly call internal APIs, and then future refactoring of this project's internals causes test failures in test assemblies that have no business in NuGet.PackageManagement's internals.

Different people in the team have different opinions, but my opinion is that we don't need to validate exact error/warning messages. If the goal is to validate that a project name, source name, or argument name exists in a log message, then `string.Contains` is sufficient in my opinion.

Maybe this feedback seems overkill for what seems like a small thing, but my entire career I've been working on 10+ years old apps, and are full of tech debt making them harder to work on. So, I'm rather sensitive to design decisions which can cause problems years later.